dags/atd_knack_esb_311.py

```diff
@@ -0,0 +1,102 @@
+import os
+
+from airflow.models import DAG
+from airflow.operators.docker_operator import DockerOperator
+from docker.types import Mount 
+from pendulum import datetime, duration, now
+
+from utils.onepassword import get_env_vars_task
+from utils.slack_operator import task_fail_slack_alert
+
+DEPLOYMENT_ENVIRONMENT = os.getenv("ENVIRONMENT", "development")
+
+DOCKER_IMAGE ="atddocker/atd-knack-311:production"
+
+DEFAULT_ARGS = {
+    "owner": "airflow",
+    "depends_on_past": False,
+    "start_date": datetime(2015, 1, 1, tz="America/Chicago"),
+    "email_on_failure": False,
+    "email_on_retry": False,
+    "retries": 0,
+    "execution_timeout": duration(minutes=5),
+    "on_failure_callback": task_fail_slack_alert,
+}
+
+REQUIRED_SECRETS_DATA_TRACKER = {
+    "KNACK_APP_ID": {
+        "opitem": "Knack AMD Data Tracker",
+        "opfield": f"production.appId",
+    },
+    "KNACK_API_KEY": {
+        "opitem": "Knack AMD Data Tracker",
+        "opfield": f"production.apiKey",
+    },
+    "ESB_ENDPOINT": {
+        "opitem": "CTM Enterprise Service Bus - ESB - 311 Interface",
+        "opfield": f"production.endpoint",
+    }
+}
+
+REQUIRED_SECRETS_SIGNS_MARKIGNS = {
+    "KNACK_APP_ID": {
+        "opitem": "Knack Signs and Markings",
+        "opfield": f"production.appId",
+    },
+    "KNACK_API_KEY": {
+        "opitem": "Knack Signs and Markings",
+        "opfield": f"production.apiKey",
+    },
+    "ESB_ENDPOINT": {
+        "opitem": "CTM Enterprise Service Bus - ESB - 311 Interface",
+        "opfield": f"production.endpoint",
+    }
+}
+
+
+with DAG(
+    dag_id=f"atd_knack_esb_311",
+    description="Publishes 311 SR activities from Knack to 311 CSR via the CTM ESB",
+    default_args=DEFAULT_ARGS,
+    schedule_interval="1-59/5 * * * *" if DEPLOYMENT_ENVIRONMENT == "production" else None,
```

Comment:
it runs every 5th minute starting at minute one—just to offset the timing from other ETLs on that timing. i dunno that is helpful or not, but that's the same schedule it uses on the legacy server.